### DISCUSSION TITLE
hide emit doesn't appear to work with sidebar

### DISCUSSION THREAD

**Discussion by plderrick** [primefaces/community#1034] [PrimeVue] [answered]
HI, 
Reasonably new to vueJS and Prime. Building an app and using sidebar component and want to track when it is dismissed. The API says it emits show and hide for open and close respectively, but doesn't appear to be working. I'm using hide for other components (e.g. Dialog) and it works fine (console log on invocation), but nothing when the sidebar is closed (see implementation below).

```
   <div v-if="getActiveOption == 'querySidebar'">
            <MapQuerySidebar @hide="setClosedOption" />
   </div>
```

I've added the OnUnmounted callback in setup to the sidebar vue, but it fires when the sidebar is reopened, not when it is dismissed.

```
        onUnmounted(() => {
            console.log("Sidebar closing");
            clearQueryMarker();
        });
```

So use case is when sidebar is closed I want to be notified that it has closed so I can do some cleanup.  Cleanest (and easiest) methods don't appear to work. 

Thanks for your help.

**Comment by tugcekucukoglu** [comment]
We could not replicate the issue. Could you share a reproducer link?

**Reply by plderrick** [reply, marked as answer]
Thanks. Looks like the emit was only caught with-in the MapQuerySidebar.vue module where SideBar was declared. I added @hide to the SideBar declaration and added a handler in MapQuerySidebar. This catches the hide as required. I then emit a new event for the high-level vue module to catch so it can do what it needs to do (see below).

```
export default {
    name: 'SideBar',
    emits: ['sideBarClosed'],
    setup(props, { emit }) {...
```

and further on

```
        const sideBarClosed = () => {
            emit("sideBarClosed");
        }
```

All works as required now.
